File: leetcode/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mini=prices[0]
        diff=0
        for i in range (len(prices)):
            if prices[i]<mini:
                mini=prices[i]
            elif prices[i]-mini>diff:
                diff=prices[i]-mini
        return diff
        

File: leetcode/0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size()<2){
            return 0;
        }
        int min_price=prices[0];
        int diff=prices[1]-prices[0];
        for (int price : prices){
            if (price<min_price){
                min_price=price;
            }
            else if (price-min_price>diff){
                diff=price-min_price;
            }
        }
        if (diff>0) return diff;
        else return 0;
    }
};
